File: articles/failure-modes-in-algorithmic-systems/prolog/failure_risk_score.pl
failure_risk_score(Likelihood, Severity, Detectability, Controllability, FailureRisk) :-
    FailureRisk is Likelihood * Severity * (1 - Detectability) * (1 - Controllability).

% Query example:
% ?- failure_risk_score(0.42, 0.86, 0.38, 0.44, FailureRisk).

File: articles/failure-modes-in-algorithmic-systems/haskell/FailureRiskScore.hs
module Main where

main :: IO ()
main = do
  let likelihood = 0.42
      severity = 0.86
      detectability = 0.38
      controllability = 0.44
      failureRisk = likelihood * severity * (1 - detectability) * (1 - controllability)
  putStrLn ("failure_risk_score=" ++ show failureRisk)
